addons/binding/org.openhab.binding.ihc/src/main/java/org/openhab/binding/ihc/internal/handler/IhcHandler.java

```diff
@@ -0,0 +1,920 @@
+                case IhcClient.CONTROLLER_STATE_INITIALIZE:
+                    logger.info("Controller state changed to initializing state, waiting for ready state");
+                    updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_STATE),
+                            new StringType("initialize"));
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+                            "Controller is in initializing state");
+                    break;
+                case IhcClient.CONTROLLER_STATE_READY:
+                    logger.info("Controller state changed to ready state");
+                    updateState(new ChannelUID(getThing().getUID(), CHANNEL_CONTROLLER_STATE), new StringType("ready"));
+                    updateStatus(ThingStatus.ONLINE);
+                    break;
+                default:
+            }
+
+            if (controllerState.equals(IhcClient.CONTROLLER_STATE_INITIALIZE)
+                    && newState.getState().equals(IhcClient.CONTROLLER_STATE_READY)) {
+                logger.debug("Reconnection request");
+                projectFile = null;
+                setReconnectRequest(true);
+            }
+        }
+
+        controllerState = newState.getState();
+    }
+
+    @Override
+    public void resourceValueUpdateReceived(WSResourceValue value) {
+        logger.debug("resourceValueUpdateReceived: {}", value);
+
+        thing.getChannels().forEach(channel -> {
+            ChannelParams params = new ChannelParams(channel);
+            if (params.getResourceId() != null && params.getResourceId().intValue() == value.getResourceID()) {
+                updateChannelState(channel, params, value);
+            }
+        });
+
+        checkPotentialButtonPresses(value);
+    }
+
+    private void updateChannelState(Channel channel, ChannelParams params, WSResourceValue value) {
+        if (params.isDirectionWriteOnly()) {
+            logger.debug("Write only channel, skip update to {}", channel.getUID());
+        } else {
+            if (params.getChannelTypeId() != null) {
+                switch (params.getChannelTypeId()) {
+                    case CHANNEL_TYPE_PUSH_BUTTON_TRIGGER:
+                        break;
+
+                    default:
+                        try {
+                            logger.debug("Update channel '{}' state, channel params: {}", channel.getUID(), params);
+                            Converter<WSResourceValue, Type> converter = ConverterFactory.getInstance()
+                                    .getConverter(value.getClass(), channel.getAcceptedItemType());
+                            State state = (State) converter.convertFromResourceValue(value,
+                                    new ConverterAdditionalInfo(null, params.isInverted()));
+                            updateState(channel.getUID(), state);
+                        } catch (NumberFormatException e) {
```

Comment:
This one might also be used with a special custom meaning, I would suggest to introduce your own exception here.

Reply:
Done